tunnel/session.go

```diff
@@ -301,13 +301,27 @@ func (s *session) Send(m *transport.Message) error {
 		// no op
 	}
 
+	// get the token
+	token, ok := m.Header["Micro-Tunnel-Token"]
```

Comment:
Token should probably be added to `t.newSession` in default so that its accessible internally as I don't think this exists in the header. It's injected after session.Send and stripped before session.Recv.